docs/source/conf.py

```diff
@@ -0,0 +1,530 @@
+        labels="labels=chore/documentation",
+    )
+
+
+def generate_colab_url(notebook_path: str) -> str:
+    notebook_path = notebook_path.replace(".txt", "")
+    notebook_path = notebook_path if notebook_path.endswith(".ipynb") else notebook_path + ".ipynb"
+    notebook_name = notebook_path.split("/")[-1]
+
+    if not is_example_notebook(notebook_name):
+        raise RuntimeError(f"Not a notebook - {notebook_path}")
+
+    template = (
+        "https://colab.research.google.com/github/{user}/{repo}/blob/{branch}/{docpath}/{notebook_path}"
+    )
+
+    return template.format(
+        user=GIT['user'],
+        repo=GIT['repo'],
+        branch=GIT['release'],
+        docpath=GIT['documentation-path'],
+        notebook_path=notebook_path
+    )
+
+
+def generate_binder_url(notebook_path: str) -> str:
+    notebook_path = notebook_path.replace(".txt", "")
+    notebook_path = notebook_path if notebook_path.endswith(".ipynb") else notebook_path + ".ipynb"
+    notebook_name = notebook_path.split("/")[-1]
+
+    if not is_example_notebook(notebook_name):
+        raise RuntimeError(f"Not a notebook - {notebook_path}")
+
+    template = (
+        "https://mybinder.org/v2/gh/{user}/{repo}/{branch}?labpath={filepath}"
+    )
+
+    return template.format(
+        user=GIT['user'],
+        repo=GIT['repo'],
+        branch=GIT['release'],
+        filepath=f"{GIT['documentation-path']}/{notebook_path}"
+    )
+
+
+@functools.lru_cache(maxsize=None)
+def get_example_notebooks() -> t.Tuple[pathlib.Path, ...]:
+    examples_folder = CURRENT_DIR / "examples"
+
+    if not examples_folder.exists() or not examples_folder.is_dir():
+        raise RuntimeError("Did not find the folder with the example notebooks.")
+
+    return tuple(it for it in examples_folder.glob("**/*.ipynb"))
+
+
+@functools.lru_cache(maxsize=None)
+def snake_case_to_camel_case(val: str) -> str:
+    return "".join(it.capitalize() for it in val.split("_") if it)
+
+
+def is_example_notebook(filepath: str) -> bool:
+    notebook_name = filepath.split("/")[-1].replace(".txt", "")
+    notebooks = {it.name for it in get_example_notebooks()}
+    return notebook_name in notebooks
+
+
+def get_check_example_api_reference(filepath: str) -> t.Optional[str]:
+
+    if not filepath.startswith("examples/checks/"):
+        return
+
+    notebook_name = snake_case_to_camel_case(
+        filepath.split("/")[-1]
+        .replace(".txt", "")
+        .replace(".ipynb", "")
+        .replace(".py", "")
+    )
+
+    import deepchecks.checks
+    check_clazz = getattr(deepchecks.checks, notebook_name, None)
+
+    if check_clazz is None or not hasattr(check_clazz, "__module__"):
+        return
+
+    apipath = f"/api/checks/generated/{check_clazz.__module__}.{notebook_name}"
+    result = f"* :doc:`API Reference - {notebook_name} <{apipath}>`"
+    return result
+
+
+
+# TODO: Not sure if we need this
+
+# class AutoAutoSummary(Autosummary):
+
+#     option_spec = {
```

Comment:
you're right we don't need this, you can remove

Reply:
removed